open-y-subprojects/openy_map
# fix: openy_map pins break when map isn't on a top level page

resolves https://github.com/ymcatwincities/openy/issues/2598

It seems like #9 was on the right track and I overcompensated with #12. This takes a happy medium.

Also, the changes in .install should resolve issues with the settings form not picking up the Locations Map icon value, as there was a mismatch between the expected key and what was getting set here. We should investigate why this was happening given that the settings in `config/install` were correct.

To test:
- install the site at a normal base path, or at a subdirectory
- run `drush updb`
- observe an option is selected for each content type's "Locations Map icon" field at `/admin/openy/settings/openy_map` and that you can save the form without making any changes.
![Map_settings___Drush_Site-Install](https://user-images.githubusercontent.com/238201/143373855-a8fa6cde-63e6-4a55-a6f4-fd9f1c19ecb0.png)
- observe the map icons at `/locations` are correct
- change the url of `/locations` to something like `/test/locations` and observe it doesn't break.

I've flagged two `@todo` items in here that could use clarification from @podarok as to whether they need changes, could be done later, or removed.
